xla/service/collective_permute_decomposer_test.cc

```diff
@@ -282,38 +276,50 @@ TEST_F(CollectivePermuteDecomposerTest, ForwardPipeline2) {
     ROOT result = u32[2] get-tuple-element(while_result), index=1
   })";
 
-  TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module,
-                          ParseAndReturnUnverifiedModule((kModuleStr)));
-  CollectivePermuteDecomposer decomposer(/*threshold_in_bytes=*/0);
-  TF_ASSERT_OK_AND_ASSIGN(bool changed, decomposer.Run(module.get()));
-  EXPECT_TRUE(changed);
-  HloInstruction* recv = FindInstruction(module.get(), "recv");
-  EXPECT_EQ(recv->channel_id().value(), 1);
-  EXPECT_THAT(recv->ToString(),
-              HasSubstr("_xla_send_recv_source_target_pairs={{3,0}}"));
-  EXPECT_THAT(recv->ToString(), HasSubstr("_xla_send_recv_pipeline=\"0\""));
-  HloInstruction* send = FindInstruction(module.get(), "send");
-  EXPECT_THAT(send->ToString(),
-              HasSubstr("_xla_send_recv_source_target_pairs={{3,0}}"));
-  EXPECT_THAT(send->ToString(), HasSubstr("_xla_send_recv_pipeline=\"0\""));
+  std::string channel_id_hlo =
+      absl::StrReplaceAll(kModuleStr, {{"[[CHANNEL_ID_1]]", "channel_id=1, "},
+                                       {"[[CHANNEL_ID_2]]", "channel_id=2, "}});
+  std::string no_channel_id_hlo = absl::StrReplaceAll(
+      kModuleStr, {{"[[CHANNEL_ID_1]]", ""}, {"[[CHANNEL_ID_2]]", ""}});
+
+  auto check_module = [](const std::string module_str,
+                         absl::flat_hash_set<int64_t> channel_ids) {
+    TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module,
+                            ParseAndReturnUnverifiedModule((module_str)));
+    CollectivePermuteDecomposer decomposer(/*threshold_in_bytes=*/0);
+    TF_ASSERT_OK_AND_ASSIGN(bool changed, decomposer.Run(module.get()));
```

Comment:
Did you consider a filecheck test here? Will leave it up to you if you want to change this.

Reply:
Sure, I can do that, but note that I have not edited any of the existing tests themselves. I just wrapped the same code within a lambda and called it with/without channel id. The diff here does not make it obvious. I will send an update soon, but please let me know if you prefer them unchanged. I prefer the filecheck tests too.